.jenkinsci-new/utils/docker-manifest.groovy

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env groovy
+
+def manifestSupportEnabled() {
+	def dockerVersion = sh(script: "docker -v", returnStdout: true).trim()
+	def experimentalEnabled = sh(script: "grep -i experimental ~/.docker/config.json", returnStatus: true)
+	return experimentalEnabled == 0 && dockerVersion ==~ /^Docker version 18.*$/
+
+}
+
+def manifestCreate(manifestListName, manifests) {
+	sh "docker manifest create ${manifestListName} ${manifests.join(' ')}"
+}
+
+def manifestAnnotate(manifestListName, manifestsWithFeatures) {
+	manifestsWithFeatures.each {
+		sh """
+			docker manifest annotate ${manifestListName} ${it['manifest']} --arch "${it['arch']}" \
```

Comment:
can we annotate with commit hash as well?

Reply:
It is docker manifest, I'm not sure if we can add our custom fields, this is need to be research and for another pr.
Also docker manifest  is experimental  feature